Formula/dotnet.rb

```diff
@@ -0,0 +1,33 @@
+class Dotnet < Formula
+  desc ".Net Core"
+  homepage "https://dotnet.microsoft.com/"
+  url "https://github.com/dotnet/source-build.git",
+      tag:      "v3.1.108-SDK",
+      revision: "05a0c8f6b13fcbec4b441c977b639b75e8a74f67"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "curl" => :build
+  depends_on "pkg-config" => :build
+  depends_on "icu4c"
+  depends_on "openssl"
+
+  def install
+    # Arguments needed to not artificially time-limit downloads from Azure.
+    # See the following GitHub issue comment for details:
+    # https://github.com/dotnet/source-build/issues/1596#issuecomment-670995776
+    reference_timeout = "/p:DownloadSourceBuildReferencePackagesTimeoutSeconds=N/A"
+    artifact_timeout = "/p:DownloadSourceBuiltArtifactsTimeoutSeconds=N/A"
+    system "./build.sh", reference_timeout, artifact_timeout
+
+    lib.mkpath
+    tarball = Dir["artifacts/*/Release/dotnet-sdk-#{version}-*.tar.gz"].first
+    system "tar", "-xzf", tarball, "--directory", lib
+
+    (bin/"dotnet").write_env_script lib/"dotnet", DOTNET_ROOT: ENV.fetch("DOTNET_ROOT", lib)
```

Comment:
Wouldn't this end up with the same result?
```suggestion
    (bin/"dotnet").write_env_script lib/"dotnet", DOTNET_ROOT:  lib
```

Reply:
Yes, we don't allow passing through arbitrary ENVs here.